pkg/apiserver/server.go

```diff
@@ -90,20 +90,18 @@ func Start(params *APIServerParams) {
 		panic(err)
 	}
 
-	if !util.IsHelmManaged() {
-		client := &client.Client{
-			TargetNamespace:   util.AppNamespace(),
-			ExistingInformers: map[string]bool{},
-			HookStopChans:     []chan struct{}{},
-		}
-		store := store.GetStore()
-		op := operator.Init(client, store, params.AutocreateClusterToken)
-		if err := op.Start(); err != nil {
-			log.Println("error starting the operator")
-			panic(err)
-		}
-		defer op.Shutdown()
+	client := &client.Client{
+		TargetNamespace:   util.AppNamespace(),
+		ExistingInformers: map[string]bool{},
+		HookStopChans:     []chan struct{}{},
+	}
+	opStore := store.GetStore()
+	op := operator.Init(client, opStore, params.AutocreateClusterToken)
```

Comment:
opStore cannot be used in helm mode.  It will attempt to connect to postgres, which works in dev env, but won't work when deployed.

Reply:
perhaps I should only initialize it when not running in helm managed mode?  the store doesnt get used for helm managed in this PR

Reply:
You have to call `operator.Init` with valid parameters.  That code does not check if store is nil.  Either it needs to be updated to not use store or operator should be turned into an interface and two different implementations, one for helm and one for normal installs.

Reply:
will work to go the interface route

Reply:
wouldn't this nil panic?
https://github.com/replicatedhq/kots/pull/3313/commits/d6bb59047d261a17679af19d71525cbbb7a9f824#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR427

`o.client` also uses store.

How about not try to fit this into existing operator code?

Reply:
it exits before that line is hit, it seems the interface route is the best option here